org.adempiere.base/src/org/compiere/print/DataEngine.java

```diff
@@ -1064,15 +1065,9 @@ else if (m_group.isFunctionColumn(pdc.getAD_PrintFormatItem_ID(), functions[f]))
 										displayType = getDisplayType(value);
 									
 										if (display.startsWith("@SCRIPT")) {
-											display = display.replace("@SCRIPT", "");
-											value = parseVariable(display, pdc, pd);
-											Interpreter bsh = new Interpreter ();
-											try {
-												value = bsh.eval(value.toString());
-											}
-											catch (EvalError e) {
-												log.severe(e.getMessage());
-											}
+											value = display.replace("@SCRIPT", "");
+											if(!scriptColumns.contains(pdc))
+												scriptColumns.add(pdc);
 											displayType = getDisplayType(value);
```

Comment:
useless to call getDisplayType here, just use a dummy one.

Reply:
I chose DisplayType.Text to be in sync with the getDisplayType(value) method